File: Task/15-puzzle-solver/Python/15-puzzle-solver-3.py
"""

Runs one test of the solver passing a
start and goal position.

"""

from astar import *
import time

# Rosetta Code start position


start_tiles =    [[ 15, 14,  1,  6],
                  [ 9, 11,  4, 12],
                  [ 0, 10,  7,  3],
                  [13,  8,  5,  2]]

goal_tiles =        [[ 1,  2,  3,  4],
                     [ 5,  6,  7,  8],
                     [ 9, 10, 11, 12],
                     [13, 14, 15,  0]]


before = time.perf_counter()

result = a_star(start_tiles,goal_tiles)

after = time.perf_counter()

print(" ")
print("Path length = "+str(len(result) - 1))
print(" ")
print("Path using rlud:")
print(" ")
print(path_as_0_moves(result))
print(" ")
print("Run time in seconds: "+str(after - before))

File: Task/15-puzzle-solver/Nim/15-puzzle-solver-3.nim
let start = getTime()
var solver = initSolver([Value  0, 12,  9, 13,
                               15, 11, 10, 14,
                                3,  7,  2,  5,
                                4,  8,  6,  1])
solver.run()
echo fmt"Execution time: {(getTime() - start).toString}."
